compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -550,8 +561,16 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling {
             case OrType(tp1, tp2) => isNullable(tp1) || isNullable(tp2)
             case _ => false
           }
-          (tp1.symbol eq NothingClass) && tp2.isValueTypeOrLambda ||
-          (tp1.symbol eq NullClass) && isNullable(tp2)
+          def isPhantom(tp: Type): Boolean = {
+            // note that the only phantom classes are PhantomAnyClass and PhantomNothingClass
```

Comment:
This was an optimisation based on the the existence of local `myPhantomAnyClass`.  I will remove it and replace it with `tp.isPhantom` which dose the same as `tp.topType != def.AnyType`.